src/config.c

```diff
@@ -117,44 +582,72 @@ int ReadConfig(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     }
 
-    if (!ValidateChunkSize(ctx, Chunk_SIZE_BYTES_SECS)) {
+    if (!ValidateChunkSize(ctx, Chunk_SIZE_BYTES_SECS, NULL)) {
         return TSDB_ERROR;
     }
     TSGlobalConfig.chunkSizeBytes = Chunk_SIZE_BYTES_SECS;
-    if (ParseChunkSize(ctx, argv, argc, "CHUNK_SIZE_BYTES", &TSGlobalConfig.chunkSizeBytes) !=
-        REDISMODULE_OK) {
+    bool chunkSizeBytesFound = false;
+    if (ParseChunkSize(ctx,
+                       argv,
+                       argc,
+                       "CHUNK_SIZE_BYTES",
+                       &TSGlobalConfig.chunkSizeBytes,
+                       &chunkSizeBytesFound) != REDISMODULE_OK) {
         RedisModule_Log(ctx, "warning", "Unable to parse argument after CHUNK_SIZE_BYTES");
         return TSDB_ERROR;
     }
+
     RedisModule_Log(ctx,
                     "notice",
                     "loaded default CHUNK_SIZE_BYTES policy: %lld",
                     TSGlobalConfig.chunkSizeBytes);
 
+    LOG_DEPRECATED_OPTION(
+        "CHUNK_SIZE_BYTES", "ts-chunk-size-bytes", chunkSizeBytesFound && showDeprecationWarning);
+
+    isDeprecated = isDeprecated || chunkSizeBytesFound;
+
     TSGlobalConfig.duplicatePolicy = DEFAULT_DUPLICATE_POLICY;
-    if (ParseDuplicatePolicy(
-            ctx, argv, argc, DUPLICATE_POLICY_ARG, &TSGlobalConfig.duplicatePolicy) != TSDB_OK) {
+    bool duplicatePolicyFound = false;
+    if (ParseDuplicatePolicy(ctx,
+                             argv,
+                             argc,
+                             DUPLICATE_POLICY_ARG,
+                             &TSGlobalConfig.duplicatePolicy,
+                             &duplicatePolicyFound) != TSDB_OK) {
         RedisModule_Log(ctx, "warning", "Unable to parse argument after DUPLICATE_POLICY");
         return TSDB_ERROR;
     }
+
+    isDeprecated = isDeprecated || duplicatePolicyFound;
+
+    LOG_DEPRECATED_OPTION(
+        "DUPLICATE_POLICY", "ts-duplicate-policy", duplicatePolicyFound && showDeprecationWarning);
+
     RedisModule_Log(ctx,
                     "notice",
                     "loaded server DUPLICATE_POLICY: %s",
                     DuplicatePolicyToString(TSGlobalConfig.duplicatePolicy));
 
     if (argc > 1 && (RMUtil_ArgIndex("ENCODING", argv, argc) >= 0 ||
                      RMUtil_ArgIndex("CHUNK_TYPE", argv, argc) >= 0)) {
+        isDeprecated = true;
+
+        if (RMUtil_ArgIndex("ENCODING", argv, argc) >= 0) {
+            LOG_DEPRECATED_OPTION("ENCODING", "ts-encoding", showDeprecationWarning);
+        }
+
         if (RMUtil_ArgIndex("CHUNK_TYPE", argv, argc) >= 0) {
-            RedisModule_Log(
-                ctx,
-                "warning",
-                "CHUNK_TYPE configuration was deprecated and will be removed in future "
-                "versions of RedisTimeSeries. Please use ENCODING configuration instead.");
+            LOG_DEPRECATED_OPTION("CHUNK_TYPE", "ts-encoding", showDeprecationWarning);
+
+            RedisModule_Log(ctx,
```

Comment:
does this warning add anything more than the previous `LOG_DEPRECATED_OPTION` already did?

Reply:
This message has been there before if you take a look. So, I left it for backward compatibility.

Reply:
Yes, I know. I see it. But the prior message was simply doing the same thing that `LOG_DEPRECATED_OPTION` is now doing. Adding another case with both `CHUNK_TYPE` and `ENCODING` together as deprecated configuration options is redundant imo. Both of those already individually warn that it is deprecated.

Reply:
Can't disagree.